core/chains/evm/gas/rollups/l1_oracle.go

```diff
@@ -47,15 +49,33 @@ func IsRollupWithL1Support(chainType chaintype.ChainType) bool {
 	return slices.Contains(supportedChainTypes, chainType)
 }
 
-func NewL1GasOracle(lggr logger.Logger, ethClient l1OracleClient, chainType chaintype.ChainType) (L1Oracle, error) {
+func NewL1GasOracle(lggr logger.Logger, ethClient l1OracleClient, chainType chaintype.ChainType, daOracle evmconfig.DAOracle) (L1Oracle, error) {
 	if !IsRollupWithL1Support(chainType) {
 		return nil, nil
 	}
 	var l1Oracle L1Oracle
 	var err error
+	if daOracle != nil {
+		switch daOracle.OracleType() {
+		case toml.OPStack:
+			l1Oracle, err = NewOpStackL1GasOracle(lggr, ethClient, chainType, daOracle)
+		case toml.Arbitrum:
+			l1Oracle, err = NewArbitrumL1GasOracle(lggr, ethClient)
+		case toml.ZKSync:
+			l1Oracle = NewZkSyncL1GasOracle(lggr, ethClient)
+		default:
+			lggr.Warnf("Unsupported DA oracle type %s. Going forward all chain configs should specify an oracle type", daOracle.OracleType())
```

Comment:
Wouldn't it be a better option to set an error and skip the l1Oracle check later?
```go
switch oracleType := daOracle.OracleType() {
// ...
default: 
     err := fmt.Errorf("no supported DA oracle type %w", oracleType
//.. 
```

Reply:
@mateusz-sekara I guess it depends on how flexible we want the fallback logic to be. If we want to be extra forgiving, we don't want to `err` out and return early here so we can still fall back to checking `chainType` below. However, it's probably best to let the dev know that their `DAOracle` config was misconfigured, so I'll take your suggestion 👍